File: Non Linear Data Structures/Trees/Python/Tree Traversals/DFS/inorder.py


class Node :
    def __init__(self , data):
        self.data = data
        self.left = None
        self.right = None
        
def inOrderTraversal(root) :
    if root == None:
        return
    inOrderTraversal(root.left)
    print(root.data)
    inOrderTraversal(root.right)

root = Node(10)
root.left = Node(11)
root.right = Node(22)

# print(root.data)
# print(root.left.data)
# print(root.right.data)

inOrderTraversal(root)

File: Non Linear Data Structures/Trees/C++/Tree Traversals/DFS/inorder.cpp
#include<iostream>
#include<stack>
using namespace std;

class Node {
    public :
        int data;
        Node* left;
        Node* right;

    Node(int data) {
        this->data = data;
        this->left = NULL;
        this->right = NULL;
    }
};

void inOrderTraversal(Node* root) {
    if(root == NULL) return;

    inOrderTraversal(root->left);
    cout<<root->data<<"\t";
    inOrderTraversal(root->right);

}


int main () {

    Node* root = new Node(10);
    root->left = new Node(11);
    root->right = new Node(21);

    inOrderTraversal(root);


    return 0;
}